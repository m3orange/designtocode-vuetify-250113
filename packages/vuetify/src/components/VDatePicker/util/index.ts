import { createItemTypeNativeListeners, createItemTypeListeners } from './eventHelpers'
import createNativeLocaleFormatter from './createNativeLocaleFormatter'
import monthChange from './monthChange'
import pad from './pad'

export {
  createItemTypeNativeListeners,
  createNativeLocaleFormatter,
  createItemTypeListeners,
  monthChange,
  pad,
}
