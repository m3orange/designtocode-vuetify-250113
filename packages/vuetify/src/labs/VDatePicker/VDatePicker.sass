@use '../../styles/tools'

.v-date-picker__input
  padding-top: 16px
  padding-left: 24px
  padding-right: 24px

.v-date-picker
  @include tools.elevation(2)
  @include tools.rounded(4px)

  overflow: hidden
  width: 328px

  .v-date-picker-month
    padding-left: 12px
    padding-right: 12px

    .v-date-picker-month__day
      height: 40px
      width: 40px

      .v-btn
        --v-btn-height: 24px
        --v-btn-size: 0.85rem
