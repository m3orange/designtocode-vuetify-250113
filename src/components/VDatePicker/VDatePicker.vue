<script>
  import { createRange } from '../../util/helpers'

  import Picker from '../../mixins/picker'
  import DateYears from './mixins/date-years'
  import DateTitle from './mixins/date-title'
  import DateHeader from './mixins/date-header'
  import DateTable from './mixins/date-table'
  import MonthTable from './mixins/month-table'
  import VBtn from '../VBtn'
  import VCard from '../VCard'
  import VIcon from '../VIcon'

  import Touch from '../../directives/touch'

  const createDefaultDateFormat = pickMonth => val => new Date(val).toISOString().substr(0, pickMonth ? 7 : 10)

  export default {
    name: 'v-date-picker',

    components: {
      VBtn,
      VCard,
      VIcon
    },

    mixins: [Picker, DateYears, DateTitle, DateHeader, DateTable, MonthTable],

    directives: { Touch },

    data () {
      return {
        tableDate: new Date(),
        originalDate: this.value,
        currentDay: null,
        currentMonth: null,
        currentYear: null,
        isReversing: false,
        narrowDays: [],
        activePicker: this.pickMonth ? 'MONTH' : 'DATE'
      }
    },

    props: {
      locale: {
        type: String,
        default: 'en-us'
      },
      dateFormat: {
        type: Function,
        default: null
      },
      titleDateFormat: {
        type: Object,
        default: null
      },
      headerDateFormat: {
        type: Object,
        default: () => ({ month: 'long', year: 'numeric' })
      },
      formattedValue: {
        required: false
      },
      allowedDates: {
        type: [Array, Object, Function],
        default: () => (null)
      },
      firstDayOfWeek: {
        type: [String, Number],
        default: 0
      },
      pickMonth: Boolean,
      yearIcon: String
    },

    computed: {
      firstAllowedDate () {
        // const date = new Date()
        // date.setDate(1)
        // date.setHours(12, 0, 0, 0)

        // if (this.allowedDates) {
        //   for (let month = 0; month < 12; month++) {
        //     const valid = date.setMonth(month)
        //     if (this.isAllowed(valid)) return valid
        //   }
        // }

        // return date
        const date = new Date()
        date.setHours(12, 0, 0, 0)

        if (this.allowedDates) {
          const millisecondOffset = 1 * 24 * 60 * 60 * 1000
          const valid = new Date(date)
          for (let i = 0; i < 31; i++) {
            if (this.isAllowed(valid)) return valid

            valid.setTime(valid.getTime() + millisecondOffset)
          }
        }

        return date
      },
      inputDate: {
        get () {
          if (!this.value) return this.firstAllowedDate
          if (this.value instanceof Date) return this.value
          if (!isNaN(this.value) ||
              typeof this.value === 'string' && this.value.indexOf(':') !== -1
          ) return new Date(this.value)

          return new Date(`${this.value}T12:00:00`)
        },
        set (val) {
          const pickerDateFormat = createDefaultDateFormat(this.pickMonth)
          this.$emit('input', val ? pickerDateFormat(val) : this.originalDate)
          this.$emit('update:formattedValue', (this.dateFormat || pickerDateFormat)(val || this.originalDate))
        }
      },
      day () {
        return this.inputDate.getDate()
      },
      month () {
        return this.inputDate.getMonth()
      },
      year () {
        return this.inputDate.getFullYear()
      },
      tableMonth () {
        return this.tableDate.getMonth()
      },
      tableYear () {
        return this.tableDate.getFullYear()
      },
      computedTransition () {
        return this.isReversing ? 'tab-reverse-transition' : 'tab-transition'
      },
      titleText () {
        let date = new Date(this.year, this.month, this.day, 1 /* Workaround for #1409 */)

        const defaultTitleDateFormat = this.pickMonth ? {
          month: 'long'
        } : {
          weekday: 'short',
          month: 'short',
          day: 'numeric'
        }
        date = date.toLocaleString(this.locale, this.titleDateFormat || defaultTitleDateFormat)

        if (this.landscape) {
          if (date.indexOf(',') > -1) date = date.replace(',', ',<br>')
          else if (date.indexOf(' ') > -1) date = date.replace(' ', '<br>')
        }

        return date
      }
    },

    watch: {
      activePicker (val, prev) {
        val === 'YEAR' && setTimeout(() => {
          this.$refs.years.scrollTop = this.$refs.years.scrollHeight / 2 - 125
        }, 350)
      },
      tableDate (val, prev) {
        this.isReversing = val < prev
      },
      value (val) {
        if (val) this.tableDate = this.inputDate
      },
      pickMonth (val) {
        if (val && this.activePicker === 'DATE') {
          this.activePicker = 'MONTH'
        }
      }
    },

    methods: {
      intifyDate(date) {
          // if (!date) return null
          // date = new Date(date)
          // return (date.getFullYear() * 12 + date.getMonth()) * 32 + 1
        if (!date) return null
        date = new Date(date)
        return (date.getFullYear() * 12 + date.getMonth()) * 32 + date.getDate()
      },

      getInputDateForYear (year) {
        let tableMonth = this.tableMonth + 1
        let day = this.day
        tableMonth = tableMonth < 10 ? `0${tableMonth}` : tableMonth
        day = day < 10 ? `0${day}` : day

        return `${year}-${tableMonth}-${day}`
      },
      save () {
        if (this.originalDate) {
          this.originalDate = this.value
        } else {
          this.originalDate = this.inputDate
        }

        if (this.$parent && this.$parent.isActive) this.$parent.isActive = false
      },
      cancel () {
        this.inputDate = this.originalDate
        if (this.$parent && this.$parent.isActive) this.$parent.isActive = false
      },
      isAllowed (date) {
        if (!this.allowedDates) return true

        if (Array.isArray(this.allowedDates)) {
          date = this.intifyDate(date)
          return !!this.allowedDates.find(allowedDate => this.intifyDate(allowedDate) === date)
        } else if (this.allowedDates instanceof Function) {
          return this.allowedDates(date)
        } else if (this.allowedDates instanceof Object) {
          date = this.intifyDate(date)
          const min = this.intifyDate(this.allowedDates.min)
          const max = this.intifyDate(this.allowedDates.max)
          return (!min || min <= date) && (!max || max >= date)
        }

        return true
      },
      genTableTouch (touchCallback) {
        return {
          name: 'touch',
          value: {
            left: e => (e.offsetX < -15) && touchCallback(1),
            right: e => (e.offsetX > 15) && touchCallback(-1)
          }
        }
      },
      genTable (tableChildren, touchCallback) {
        const options = {
          staticClass: 'picker--date__table',
          'class': {
            'picker--month__table': this.activePicker === 'MONTH'
          },
          on: this.scrollable ? { wheel: this.monthWheelScroll } : undefined,
          directives: [this.genTableTouch(touchCallback)]
        }

        const table = this.$createElement('table', {
          key: this.activePicker === 'MONTH' ? this.tableYear : this.tableMonth
        }, tableChildren)

        return this.$createElement('div', options, [
          this.$createElement('transition', {
            props: { name: this.computedTransition }
          }, [table])
        ])
      },
      genPickerBody (h) {
        const pickerBodyChildren = []
        if (this.activePicker === 'DATE') {
          pickerBodyChildren.push(h('div', { staticClass: 'picker--date__header' }, [this.genSelector()]))
          pickerBodyChildren.push(this.genTable([
            this.dateGenTHead(),
            this.dateGenTBody()
          ], value => this.tableDate = new Date(this.tableYear, this.tableMonth + value)))
        } else if (this.activePicker === 'MONTH') {
          pickerBodyChildren.push(h('div', { staticClass: 'picker--date__header' }, [this.genSelector()]))
          pickerBodyChildren.push(this.genTable([
            this.monthGenTBody()
          ], value => this.tableDate = new Date(this.tableYear + value, 0)))
        } else if (this.activePicker === 'YEAR') {
          pickerBodyChildren.push(this.genYears(year => this.getInputDateForYear(year)))
        }

        return pickerBodyChildren
      },
      renderPicker (h) {
        const children = []

        !this.noTitle && children.push(this.genTitle(this.titleText))

        children.push(h('transition', {
          props: {
            origin: 'center center',
            mode: 'out-in',
            name: 'scale-transition'
          }
        }, [h('div', {
          staticClass: 'picker__body',
          key: this.activePicker
        }, this.genPickerBody(h))]))

        this.$scopedSlots.default && children.push(this.genSlot())

        return h('v-card', {
          staticClass: 'picker picker--date',
          'class': {
            'picker--landscape': this.landscape,
            ...this.themeClasses
          }
        }, children)
      }
    },

    created () {
      const date = new Date()
      date.setDate(date.getDate() - date.getDay() + parseInt(this.firstDayOfWeek))

      createRange(7).forEach(() => {
        const narrow = date.toLocaleString(this.locale, { weekday: 'narrow' })
        this.narrowDays.push(narrow)

        date.setDate(date.getDate() + 1)
      })

      this.tableDate = this.inputDate
    },

    render (h) {
      return this.renderPicker(h)
    },

    mounted () {
      const date = new Date()
      this.currentDay = date.getDate()
      this.currentMonth = date.getMonth()
      this.currentYear = date.getFullYear()
    }

  }
</script>

<style lang="stylus" src="../../stylus/components/_pickers.styl"></style>
<style lang="stylus" src="../../stylus/components/_date-picker.styl"></style>
