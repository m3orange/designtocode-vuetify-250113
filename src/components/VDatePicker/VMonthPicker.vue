<script>
  import { createRange } from '../../util/helpers'

  import Picker from '../../mixins/picker'
  import CalendarPicker from './mixins/calendar-picker'
  import CalendarYears from './mixins/calendar-years'
  import CalendarTitle from './mixins/calendar-title'
  import CalendarHeader from './mixins/calendar-header'
  import MonthTable from './mixins/month-table'
  import VBtn from '../VBtn'
  import VCard from '../VCard'
  import VIcon from '../VIcon'

  import Touch from '../../directives/touch'

  const defaultDateFormat = val => new Date(val).toISOString().substr(0, 7)

  export default {
    name: 'v-month-picker',

    components: {
      VBtn,
      VCard,
      VIcon
    },

    mixins: [Picker, CalendarPicker, CalendarYears, CalendarTitle, CalendarHeader, MonthTable],

    directives: { Touch },

    data () {
      return {
        tableDate: new Date(),
        originalDate: this.value,
        currentDay: null,
        currentMonth: null,
        currentYear: null,
        isSelected: false,
        isReversing: false,
        narrowDays: []
      }
    },

    props: {
      locale: {
        type: String,
        default: 'en-us'
      },
      dateFormat: {
        type: Function,
        default: defaultDateFormat
      },
      formattedValue: {
        required: false
      },
      allowedDates: {
        type: [Array, Object, Function],
        default: () => (null)
      },
      yearIcon: String
    },

    computed: {
      firstAllowedDate () {
        const date = new Date().setDate(1).setHours(12, 0, 0, 0)

        if (this.allowedDates) {
          for (let month = 0; month < 12; month++) {
            const valid = date.setMonth(month)
            console.log('valid', valid)
            if (this.isAllowed(valid)) return valid
          }
        }

        return date
      },
      inputDate: {
        get () {
          if (!this.value) return this.firstAllowedDate
          if (this.value instanceof Date) return this.value
          if (!isNaN(this.value) ||
              typeof this.value === 'string' && this.value.indexOf(':') !== -1
          ) return new Date(this.value)

          return new Date(`${this.value}T12:00:00`)
        },
        set (val) {
          this.$emit('input', val ? defaultDateFormat(val) : this.originalDate)
          this.$emit('update:formattedValue', val ? this.dateFormat(val) : this.dateFormat(this.originalDate))
        }
      },
      titleText () {
        let date = new Date(this.year, this.month, 1, 12 /* Workaround for #1409 */)

        return date.toLocaleString(this.locale, { month: 'long' }).trim()
      }
    },

    methods: {
      intifyDate(date) {
          if (!date) return null
          date = new Date(date)
          return (date.getFullYear() * 12 + date.getMonth()) * 32 + 1
      },
      getInputDateForYear (year) {
        let month = this.month + 1
        month = month < 10 ? `0${month}` : month

        return `${year}-${month}`
      }
    },

    created () {
      this.tableDate = this.inputDate
    },

    render (h) {
      const headerSelector = this.selected ? null : this.genSelector(this.tableYear,
        new Date(this.tableYear, 0, 1, 1 /* Workaround for #1409 */)
        .toLocaleString(this.locale, { year: 'numeric' }),
        change => new Date(change, 0))
      return this.renderPicker(h, 'picker picker--date picker--month', headerSelector)
    }
  }
</script>

<style lang="stylus" src="../../stylus/components/_pickers.styl"></style>
<style lang="stylus" src="../../stylus/components/_date-picker.styl"></style>
