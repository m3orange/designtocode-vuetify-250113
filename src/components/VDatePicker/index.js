import VDatePicker from './VDatePicker'

export { VDatePicker }

VDatePicker.install = function install (Vue) {
  Vue.component(VDatePicker.name, VDatePicker)
}

export default VDatePicker
