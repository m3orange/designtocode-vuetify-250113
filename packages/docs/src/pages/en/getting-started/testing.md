---
nav: Testing
meta:
  title: Testing
  description: Details for v3 beta release - faq, changes, and upgrading.
  keywords: test, testing, vitest, jest, vue 3, vuetify 3
---

<script setup>
  import { ref } from 'vue'

  const tab = ref('testing-library')
</script>

# Testing

Introduction here

## End-to-end testing

### Cypress

TBW

## Unit testing

We recommend [vitest](https://vitest.dev/) for your unit testing needs, if you are running a vite based project.

### Vitest

Start by installing `vitest` and the `jsdom` environment.

```bash
yarn -D vitest jsdom
```

Then configure vitest using the `test` object in the vite config. Make sure to add `vuetify` as an inline dependency.

```ts { data-resource="vite.config.ts" }
...

export default defineConfig({
  ...,
  test: {
    globals: true,
    environment: 'jsdom',
    deps: {
      inline: ['vuetify']
    }
  }
})
```

That should be enough to run tests. Go to the [Writing tests](#writing-tests) section to see how to render and test Vuetify components

### Jest

If you already have an existing jest setup that you want to continue using, have a look at [vite-jest](https://github.com/sodatea/vite-jest), which offers first-class Vite integration for jest.

### Writing tests

Here is a simple example component with a **VBtn** component.

```html { data-resource="Hello.vue" }
<script setup lang="ts">
    import { computed, ref } from 'vue'
    import { VBtn } from 'vuetify/components'

    const props = defineProps<{ count: number }>()
    const times = ref(2)
    const result = computed(() => props.count * times.value)

    defineExpose(props)
</script>

<template>
    <div>{{ count }} x {{ times }} = {{ result }}</div>
    <VBtn @click="times += 1">
    x1
    </VBtn>
</template>
```

To render Vue components in unit tests, you can use either [@testing-library/vue](https://github.com/testing-library/vue-testing-library) or [@vue/test-utils](https://github.com/vuejs/test-utils). You can find a good comparison of the two in the [Vue documentation](https://vuejs.org/guide/scaling-up/testing.html#mounting-libraries).

There is not much difference between the two when it comes to mounting Vuetify components. In the examples below we have created a helper function that registers Vuetify as a global plugin.

<v-tabs v-model="tab" color="primary">
  <v-tab value="testing-library" variant="plain">@testing-library/vue</v-tab>
  <v-tab value="test-utils" variant="plain">@vue/test-utils</v-tab>
</v-tabs>

<v-window v-model="tab">
  <v-window-item value="testing-library">

```ts { data-resource="Hello.spec.ts" }
import { render, fireEvent } from '@testing-library/vue'
import { test, expect } from 'vitest'
import { createVuetify } from 'vuetify'
import Hello from './Hello.vue'

function renderWithVuetify (component, options) {
    const vuetify = createVuetify()
    return render(component, {
        ...options,
        global: {
            plugins: [vuetify]
        },
    })
}

test('mount component', async () => {
  expect(Hello).toBeTruthy()

  const wrapper = renderWithVuetify(Hello, {
    props: {
      count: 4,
    },
  })

  expect(wrapper.queryByText('4 x 2 = 8')).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()

  const button = wrapper.queryByText('x1')!

  await fireEvent.click(button)

  expect(wrapper.queryByText('4 x 3 = 12')).toBeTruthy()

  await fireEvent.click(button)

  expect(wrapper.queryByText('4 x 4 = 16')).toBeTruthy()
})
```

  </v-window-item>
  <v-window-item value="test-utils">

```ts { data-resource="Hello.spec.ts" }
import { mount } from '@vue/test-utils'
import { test, expect } from 'vitest'
import { createVuetify } from 'vuetify'
import Hello from './Hello.vue'

function mountWithVuetify (component, options) {
    const vuetify = createVuetify()
    return mount(component, {
        ...options,
        global: {
            plugins: [vuetify]
        },
    })
}

test('mount component', async () => {
  expect(Hello).toBeTruthy()

  const wrapper = mountWithVuetify(Hello, {
    props: {
      count: 4,
    },
  })

  expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('.v-btn').trigger('click')

  expect(wrapper.text()).toContain('4 x 3 = 12')

  await wrapper.get('.v-btn').trigger('click')

  expect(wrapper.text()).toContain('4 x 4 = 16')
})
```

  </v-window-item>
</v-window>





