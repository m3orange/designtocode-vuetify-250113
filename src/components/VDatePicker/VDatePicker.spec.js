import VDatePicker from '~components/VDatePicker'
import { test } from '~util/testing'

test('VDatePicker.js', ({ mount }) => {
  it('should display the correct date in title and header', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: new Date('November 1 2005')
      }
    })

    const title = wrapper.find('.picker--date__title-date div')[0]
    const header = wrapper.find('.picker--date__header-selector-date strong')[0]

    expect(title.text()).toBe('Tue, Nov 1')
    expect(header.text()).toBe('November 2005')
  })

  it('should match snapshot with default settings', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with pick-month prop', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        type: 'month'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with dark theme', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        dark: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        allowedDates: { min: '2013-05-03', max: '2013-05-19' }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates and pick-month prop', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: ['2013-01', '2013-03', '2013-05', '2013-07', '2013-09']
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with no title', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        noTitle: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with first day of week', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        firstDayOfWeek: 2
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
