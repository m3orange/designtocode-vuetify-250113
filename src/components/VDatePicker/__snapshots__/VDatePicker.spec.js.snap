// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDatePicker.js should match snapshot with allowed dates 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with allowed dates and pick-month prop 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with colored date picker 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with colored date picker 2`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with colored month picker 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with colored month picker 2`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with dark theme 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with default settings 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with first day of week 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with locale 1`] = `

<div class="picker picker--date card"
     style="height: auto;"
>
  <div class="picker__title">
    <div class="picker--date__title-year">
      2013
    </div>
    <div class="picker--date__title-date active">
      <div>
        M05 7, Tue
      </div>
    </div>
  </div>
  <div class="picker__body">
    <div class="picker--date__header">
      <div class="picker--date__header-selector">
        <button type="button"
                class="btn btn--icon btn--raised"
        >
          <div class="btn__content">
            <i class="material-icons icon">
              chevron_left
            </i>
          </div>
        </button>
        <div class="picker--date__header-selector-date">
          <strong>
            2013 M05
          </strong>
        </div>
        <button type="button"
                class="btn btn--icon btn--raised"
        >
          <div class="btn__content">
            <i class="material-icons icon">
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
    <div class="picker--date__table">
      <table>
        <thead>
          <tr>
            <th>
              S
            </th>
            <th>
              M
            </th>
            <th>
              T
            </th>
            <th>
              W
            </th>
            <th>
              T
            </th>
            <th>
              F
            </th>
            <th>
              S
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  1
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  2
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  3
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  4
                </span>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  5
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  6
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat btn--active"
              >
                <span class="btn__content">
                  7
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  8
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  9
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  10
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  11
                </span>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  12
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  13
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  14
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  15
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  16
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  17
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  18
                </span>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  19
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  20
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  21
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  22
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  23
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  24
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  25
                </span>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  26
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  27
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  28
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  29
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  30
                </span>
              </button>
            </td>
            <td>
              <button type="button"
                      class="btn btn--date-picker btn--floating btn--small btn--flat"
              >
                <span class="btn__content">
                  31
                </span>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              &nbsp;
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>

`;

exports[`VDatePicker.js should match snapshot with month formatting functions 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with no title 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with pick-month prop 1`] = `

<!---->

`;

exports[`VDatePicker.js should match snapshot with title/header formatting functions 1`] = `

<!---->

`;
